### PR TITLE
Removed external library and added clustering algorithm

### LINKED ISSUE #345: Reduce size of jar for utbot-summary module
**Description**

Existing solution requires in utbot-summary import of ML library with some megabytes of dependencies, but in reality it uses in one place to cluster the test cases based on path executions

**Expected behavior**

Custom implementation of dbscan or hdbscan or another clustering algorithm

### PR DESCRIPTION
**Pull request by amandelpie**
# Description

The idea behind this PR is replacement of the clustering algorithm from the external library on the our own implementation to reduce the size of jar

Fixes #345 

## Type of Change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

## Automated Testing

All utbot-summaries tests should generate the same results

# Checklist (remove irrelevant options):

- [x] The change followed the style guidelines of the UTBot project
- [x] Self-review of the code is passed
- [x] The change contains enough commentaries, particularly in hard-to-understand areas
- [x] New documentation is provided or existed one is altered
- [x] No new warnings
- [x] Tests that prove my change is effective
- [x] All tests pass locally with my changes

### CODE REVIEW

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/UtSettings.kt [hunk 1]
```diff
@@ -260,7 +260,7 @@ object UtSettings {
     /**
      * Set to true to start fuzzing if symbolic execution haven't return anything
      */
-    var useFuzzing: Boolean by getBooleanProperty(true)
```
Comment (Markoutte) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/UtSettings.kt:263]: Is it intended?

Comment (amandelpie) [reply utbot-framework-api/src/main/kotlin/org/utbot/framework/UtSettings.kt:263]: Nope

File: utbot-summary/src/main/kotlin/org/utbot/summary/clustering/dbscan/DBSCANTrainer.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    fun fit(data: Array<T>): DBSCANModel {
+        require(data.isNotEmpty()) { "Nothing to learn, data is empty." }
+
+        if (rangeQuery is LinearRangeQuery) {
+            rangeQuery.data = data
+            rangeQuery.metric = metric
+        } // TODO: could be refactored if we add some new variants of RangeQuery
+
+        val labels = IntArray(data.size) { _ -> UNDEFINED }
+        var k = 0
```
[the remark this answers is not in the document]
Comment (amandelpie) [reply utbot-summary/src/main/kotlin/org/utbot/summary/clustering/dbscan/DBSCANTrainer.kt:40]: agree

**Review by onewhl — APPROVED** [review]
Looks good to me!

**Pull request merged**